perceval/components/source.py

```diff
@@ -21,54 +21,47 @@
-            :param context: gives a local context for source specific features, like `discernability_tag`
-            """
-
-    def __init__(self, brightness: float = 1,
-                 multiphoton_component: float = None,
-                 multiphoton_model: Literal["distinguishable", "indistinguishable"] = "distinguishable",
-                 purity: float = None,
+    We build on a phenomenological model first introduced in ref. [1] where an imperfect quantum-dot based single-photon
+    source is modeled by a statistical mixture of Fock states. The model developed here, first introduced in ref. [2],
+    constructs the input multi-photon state using features specific to Perceval.
+    [1] Pont, Mathias, et al. Physical Review X 12, 031033 (2022). https://doi.org/10.1103/PhysRevX.12.031033
+    [2] Pont, Mathias, et al. arXiv preprint arXiv:2211.15626 (2022). https://doi.org/10.48550/arXiv.2211.15626
+
+    :param emission_probability: probability that the source emits at least one photon
+    :param multiphoton_component: second order intensity autocorrelation at zero time delay :math:`g^{(2)}(0)`
+    :param indistinguishability: 2-photon mean wavepacket overlap
+    :param losses: optical losses
+    :param multiphoton_model: `distinguishable` if additional photons are distinguishable, `indistinguishable` otherwise
+    :param context: gives a local context for source specific features, like `discernability_tag`
+    """
+
+    def __init__(self, emission_probability: float = 1,
+                 multiphoton_component: float = 0,
                  indistinguishability: float = 1,
-                 indistinguishability_model: Literal["homv", "linear"] = "homv",
-                 overall_transmission: float = 1,
+                 losses: float = 0,
+                 multiphoton_model: Literal["distinguishable", "indistinguishable"] = "distinguishable",
                  context: Dict = None) -> None:
 
-        if multiphoton_component is None:
-            if purity is None:
-                multiphoton_component = 0
-            else:
-                p2 = brightness * (1 - purity)
-                p1 = brightness - p2
-                multiphoton_component = 2 * p2 / (p1 + 2 * p2) ** 2
-        else:
-            assert purity is None, "cannot set both purity and multiphoton_component"
-        assert brightness * multiphoton_component <= 0.5, "brightness * g2 higher than 0.5 can not be computed for now"
-        self.brightness = brightness
-        self.overall_transmission = overall_transmission
-        self.multiphoton_component = multiphoton_component
+        assert 0 < emission_probability <= 1, "emission_probability must be in ]0;1]"
+        assert 0 <= losses <= 1, "losses must be in [0;1]"
+        assert 0 <= multiphoton_component <= 1, "multiphoton_component must be in [0;1]"
+        assert emission_probability * multiphoton_component <= 0.5,\
+            "brightness * g2 higher than 0.5 can not be computed for now"
```

Comment:
Now that brightness doesn't exist anymore, this message must be changed

Reply:
Ok